src/benchmark_register.cc

```diff
@@ -124,6 +156,10 @@ bool BenchmarkFamilies::FindBenchmarks(
     // Family was deleted or benchmark doesn't match
     if (!family) continue;
 
+    if (auto p_fixture_benchmark = dynamic_cast<FixtureBenchmark*>(family.get())) {
```

Comment:
i think this is the first time we've had an RTTI requirement in the library.
https://google.github.io/styleguide/cppguide.html#Run-Time_Type_Information__RTTI_

ideally we'd resolve this through virtuals (though i understand this is hard given you don't want to define `TryCreateFixture` on the base class.

is it completely onerous to have a separate list of fixture benchmarks on which this can be called while also having them added to the full set of benchmarks?

Reply:
alternatively, perhaps `ApplyFunctions` can also create the fixture if necessary. ie, use a virtual `ApplyFunctions` method (maybe renamed to `SetUp` ;)) that can do more in the case of a fixture benchmark.

Reply:
I think it is good practice to add a  private virtual member function named setUp/prepare or something like this.